src/main/java/br/com/tecnodev/entities/category/api/CategoryApiDTO.java

```diff
@@ -0,0 +1,63 @@
+package br.com.tecnodev.entities.category.api;
+
+import br.com.tecnodev.entities.category.Category;
+import br.com.tecnodev.entities.subCategory.SubCategory;
+
+import java.util.List;
+
+public class CategoryApiDTO {
+
+    private final String name;
+    private final String code;
+    private final Integer orderInSystem;
+    private final String colorCode;
+    private final String studyGuide;
+    private final Long totalCourse;
+    private final List<SubCategoryApiDTO> subCategories;
+
+
+    public CategoryApiDTO(Category category) {
+        this.name = category.getName();
+        this.code = category.getCode();
+        this.orderInSystem = category.getOrderInSystem();
+        this.colorCode = category.getColorCode();
+        this.studyGuide = category.getStudyGuide();
+        this.totalCourse = category.getTotalCoursesFromCategory();
+        this.subCategories = getSubCategories(category);
+    }
+
+    private List<SubCategoryApiDTO> getSubCategories(Category category) {
+        return SubCategoryApiDTO.toDto(category.getSubCategories().stream()
+                .filter(SubCategory::isActive)
+                .toList());
+    }
```

Comment:
Só dando a sugestão aqui que eu tinha comentado:

```suggestion
        this.subCategories = SubCategoryApiDTO.toDto(category.getActiveSubcategories());
    }
```

Reply:
aah sim, sabia que estava faltando alguma alteração hehe, era isso. :sweat_smile: